### PR TITLE
GlobalAveragePooling1D supports masking

### PR DESCRIPTION
**Pull request by yanboliang**
Pick https://github.com/keras-team/keras/pull/10913 to ```tf.keras```, to make ```GlobalAveragePooling1D``` support masking.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Assignee @protoget: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yanboliang** [comment]
Gently ping @fchollet @protoget . Would mind to have a look at this? Thanks.

**Review by fchollet — DISMISSED** [review]
LGTM, thank you for the PR.

**Comment by yanboliang** [comment]
I updated test goldens(tensorflow/tools/api/golden/v1) to fix the API compatibility test. Does anyone can re-trigger the test? Thanks.

File: tensorflow/python/keras/layers/pooling.py [hunk 1]
```diff
@@ -586,8 +586,23 @@ class GlobalAveragePooling1D(GlobalPooling1D):
...
   """
 
-  def call(self, inputs):
-    return backend.mean(inputs, axis=1)
+  def __init__(self, **kwargs):
+    super(GlobalAveragePooling1D, self).__init__(**kwargs)
+    self.supports_masking = True
+
+  def call(self, inputs, mask=None):
+    if mask is not None:
```
Comment (facaiy) [inline tensorflow/python/keras/layers/pooling.py]: @martinwicke Hi, I see fchollet has approved the change, do we still need to assign API Review? Moreover, I think v2 golden files are also need to be updated, how do you think?

**Comment by tensorflowbutler** [comment]
Nagging Assignees @facaiy, @protoget: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by yanboliang** [comment]
Gently ping @protoget . Can you move forward this? Thanks.

**Comment by drpngx** [comment]
@yanboliang could you pull rebase and push again?

**Comment by yanboliang** [comment]
@drpngx I find this PR has been included in https://github.com/tensorflow/tensorflow/commit/8ef3e7c8c053cb6dad530e13c478bbd406ea2c95 . I'm closing this.
